### DISCUSSION TITLE
DataReader with Column Names

### DISCUSSION THREAD

**Discussion by rsdelapaz** [mini-software/MiniExcel#351] [Q&A]
Hi, guys,

i am using OpenXML and i have OutOfMemory problems because i am trying to export to a Excel file 4 Sheets with more than 100K.

Therefore i would like to try it with MiniExcel, but i am not sure if it's possible to use DataReader with Column Names, not with the name of the column from my SQL query.

Is this the best way to export the Excel file as i have previously described?

Thank you in advace,

Rspaz

**Comment by shps951023** [comment]
@rsdelapaz 
You can make an IEnumerable `yield` Dictionary method, [document link](https://github.com/MiniExcel/MiniExcel#5-dynamic-i18n-multi-language-and-role-authority-management), e.g.
```csharp
private IEnumerable<IDictionary<string, object>> GetData()
{
	var reader = Connection.ExecuteReader("select 1 keyId,'Jack' name");
	while (reader.Read())
	{
		yield return new Dictionary<string, object>(){
			{"ID",reader.GetValue(0)},
			{"First Name",reader.GetValue(1)},
		};
	}
}
```
![image](https://user-images.githubusercontent.com/12729184/160642387-f17d5a8f-597f-4b03-8168-9a5e6c2ec274.png)

Or you could help me to create a issue, I'll think like dynamic mapping logic to resolve the issue easily.
